pallets/parachain-staking/src/tests.rs

```diff
@@ -1079,6 +1077,241 @@ fn payouts_follow_nomination_changes() {
 		});
 }
 
+#[test]
+/// Verify that only the top T::MaxNominatorsPerCollator are in top
+/// and the bottom nominators set is unbounded
+/// - MaxNominatorsPerCollator = 4
+/// - MaxCollatorsPerNominator = 4
+fn collator_nominator_sets_behave() {
+	ExtBuilder::default()
+		.with_balances(vec![
+			(1, 100),
+			(2, 100),
+			(3, 100),
+			(4, 100),
+			(5, 100),
+			(6, 100),
+			(7, 100),
+			(8, 100),
+			(9, 100),
+			(10, 100),
+		])
+		.with_collators(vec![(1, 20), (2, 20)])
+		.with_nominations(vec![
+			(3, 1, 11),
+			(4, 1, 12),
+			(5, 1, 13),
+			(6, 1, 14),
+			(7, 2, 15),
+			(8, 2, 16),
+			(9, 2, 17),
+			(10, 2, 18),
+		])
+		.build()
+		.execute_with(|| {
+			// sanity check that 3-10 are nominators immediately
+			for i in 3..11 {
+				assert!(Stake::is_nominator(&i));
+			}
+			// ~ COLLATOR 1 NominationQ tests ~
+			let collator1_state = Stake::collator_state2(1).unwrap();
+			// 11 + 12 + 13 + 14 + 20 = 70 (top 4 + self bond)
+			assert_eq!(collator1_state.total_counted, 70);
+			assert_eq!(collator1_state.total_counted, collator1_state.total_backing);
+			// Top Nominations are full and new highest nomination is made
+			assert_ok!(Stake::nominate(Origin::signed(7), 1, 15));
+			let mut expected = vec![Event::Nomination(7, 15, 1, true, 74)];
```

Comment:
```suggestion
			let mut expected_events = Vec::new();
			expected_events.push(Event::Nomination(7, 15, 1, true, 74));
```

There are two changes here.
1. The name. expected_events makes it easier to read as you go through the test.
2. We `push` this event on. So it looks the same as all the other ones throughout the test.

Just trying to improve readability of this complex test.